### ISSUE TITLE
Have SINDyOptimizer test for optimizer attributes

### ISSUE THREAD

**Issue by briandesilva** [dynamicslab/pysindy#50]
The `SINDyOptimizer` object assumes that the optimizer that's passed in has a number of methods and properties, but we don't actually check for any of them. I think it would be better if we had a series of checks in the `__init__` function so that users would see error messages as soon as they passed in their custom optimizers rather than when they call `SINDy.fit`. It would be nice to just check if the optimizer inherits from `sklearn.linear_model._base.LinearModel`, but they are discouraging users from importing from the `_*` modules. 

Methods to check for:
- `fit`
- `predict`

Attributes to check for:
 - `fit_intercept`
- `normalize`
- `coef_`
- `intercept_`

**Comment by briandesilva** [comment]
Solved in #52
